resolver/ng/init.sls

```diff
@@ -40,3 +40,27 @@
     - onchanges:
       - file: {{ sls }}~update-resolv.conf-file
 {% endif %}
+
+    {### Prevent NetworkManager managing resolvconf ###}
+    {### avoiding https://github.com/saltstack/salt/issues/45421 ###}
+  {% if salt['file.file_exists']( resolver.ng.networkmanager.file ) %}
+
+    {% for config in resolver.ng.networkmanager.regex %}
+{{ sls }}_networkmanager_dns_{{ config[0] }}:
+  file.replace:
+    - name: {{ resolver.ng.networkmanager.file }}
+    - pattern: {{ config[1] }}
+    - repl: {{ config[2] }}
+    - flags: ['IGNORECASE', 'MULTILINE']
+    - onlyif: systemctl is-enabled {{ resolver.ng.networkmanager.service }}
+    - require:
+      - file: {{ sls }}~update-resolv.conf-file
+    - watch_in:
+      - cmd: {{ sls }}_networkmanager_dns_{{ config[0] }}
+  cmd.run:
```

Comment:
I found [this](https://github.com/saltstack/salt/issues/22467#issuecomment-270022053) maybe it help to test it without make extra part just for docker.

Reply:
I think if there is no escape of that, let's use grains to avoid system dependencies (even essential like this one):
```
{% if salt['grains.get']('virtual_subtype') == 'Docker' %}
...
{% endif %}
```

Reply:
Are you content to use grains instead.  I did some reading about getting systemd running in docker and some people were saying its quite tricky.

Reply:
I'd like to get rid of it if it's possible :D
But if we can't, then grains should be better :-)
I tested it with Ubuntu 16:04 docker image, and it reports the right value.

Reply:
I accept the challenge ;-) Leave with me.

Reply:
Cool ;)
BTW, I don't remember where, but I think all what we need here is just starting `systemd`.

so this should fix the service issue:
```
...
  driver_config:
    image: centos:7
    run_command: /usr/lib/systemd/systemd
...
```

Reply:
That is what I do, https://github.com/gtmanfred/wordpress-formula/blob/master/.kitchen.docker.yml

also what we use for saltstack/salt, saltstack/salt-jenkins and saltstack-kitchen-salt